src/auth/user.ts

```diff
@@ -0,0 +1,72 @@
+import * as env from "dotenv";
+env.config();
+
+import express from "express";
+import { prisma } from "../generated/prisma-client";
+import { OAuth2Client } from "google-auth-library";
+import { generateToken } from "../utils/authToken";
+import { TokenPayload } from "google-auth-library/build/src/auth/loginticket";
+
+const client: OAuth2Client = new OAuth2Client(process.env.GOOGLE_CLIENT_ID);
+
+const router = express.Router();
+
+async function getPayloadFromGoogle(
+  token: string
+): Promise<TokenPayload> | null {
+  try {
+    const ticket = await client.verifyIdToken({
+      idToken: token,
+      audience: process.env.GOOGLE_CLIENT_ID,
+    });
+    return ticket.getPayload();
+  } catch (e) {
+    console.log(e);
+    return null;
+  }
+}
+
+router.post("/login", async (req, res) => {
+  const token: string = req.headers.authorization;
+  const payload = await getPayloadFromGoogle(token);
+  if (payload === null) {
+    res
+      .status(401)
+      .send(
+        "Not using valid email address. Check if the email address ends with @connect.hku.hk"
+      );
+    return;
+  }
```

Comment:
this doesn't seem right, no checks for `@connect.hku.hk` were done before this error response

Reply:
I should have put this in TODO.

Something new coming up here is an environment variable for declaring the organisation domain. It makes the application a little more robust, and will be helpful in the future for Google OAuth tests

Reply:
I see. That's great